snooty/parser.py

```diff
@@ -613,6 +620,28 @@ def __init__(
         self.asset_dg: "networkx.DiGraph[FileId]" = networkx.DiGraph()
         self._expensive_operation_cache = Cache()
 
+        self.published_branches: SerializableType = self.get_parsed_branches()
+        self.backend.on_published_branches(self.prefix, self.published_branches)
+
+    def get_parsed_branches(self) -> SerializableType:
+        published_branches: Dict[str, SerializableType] = {}
+        path = self.root
+        try:
+            with path.joinpath("published-branches.yaml").open(encoding="utf-8") as f:
+                data = yaml.safe_load(f)
+                try:
+                    result = check_type(
+                        gizaparser.published_branches.PublishedBranches, data
+                    )
+                    return result.serialize()
+                except LoadError as err:
+                    raise ValueError("Error reading file: " + str(err))
+        except FileNotFoundError:
+            pass
+        except LoadError as err:
+            raise ValueError("Error loading file: " + str(err))
+        return published_branches
```

Comment:
`published_branches` is always `{}` at this point in the control flow, right?

Reply:
Indeed.